### ISSUE TITLE
Mail Content-Type fallback to US_ASCII fails

### ISSUE THREAD

**Issue by tlhackque** [vixie/cron#10]
In cron.c, line 104 we have:

```C
setlocale(LC_COLLATE, "C"); /* Except for collation, since load_database() uses a-z */
       /* Except that "US-ASCII" is preferred to "ANSI_x3.4-1968" in MIME,
        * even though "ANSI_x3.4-1968" is the official charset name. */
       if ( ( cs = nl_langinfo( CODESET ) ) != 0L && 
               strcmp(cs, "ANSI_x3.4-1968") != 0 )
           strncpy( cron_default_mail_charset, cs, MAX_ENVSTR );
       else
           strcpy( cron_default_mail_charset, "US-ASCII" );
```
But `nl_langinfo()` can return `ANSI_X3.4-1968`.  Note the upper-case 'X'.  This  results in `cron_default_mail_charset` acquiring that value instead of the intended `US-ASCII`. 

It's not clear whether any environment returns the lower-case 'x', but the simplest fix is to replace `strcmp` with the case-insensitive `strcasecmp(3)`.  (`strings.h` or `string.h`)

You can demonstrate this with the following Perl on-liner (Perl exposes the same APIs):

```sh
$  perl -MPOSIX -MI18N::Langinfo=langinfo,CODESET -e'setlocale(LC_ALL,"fr_FRX");print langinfo(CODESET),"\n";'
ANSI_X3.4-1968
$
```

In my case, on  `Linux  5.17.12-100.fc34.x86_64`, Perl `v5.32.1`; `fr_FRX` is not a defined locale.

**Comment by vixie** [comment]
where is this cron.c? the one i see in origin/master has no such code near its line 104. i'll be happy to look into this but at the moment i don't know which fork of cron's source code you are experiencing. 
<img width="391" alt="image" src="https://user-images.githubusercontent.com/4666461/199858409-74396f7d-2c5c-4d7f-bcdc-a2477c18984e.png">

**Comment by tlhackque** [comment]
Apparently I'm confused.  Not your bug.  The code in question shows up in multiple forks of cron - and when I tried to track it upstream, I thought yours.  I was wrong.  Hopefully you won't be blamed for this one by anyone else.

As for how I got it wrong: I did an initial web search for "vixie cron",  and got this:
![image](https://user-images.githubusercontent.com/332018/199860380-cb4725b5-3683-49bc-840f-b598439a3074.png)

The first link comes here, as expected.  But the second, which looks to be cron.c here - doesn't.   Appears to be Debian.  I missed the fine print when I came back for a second pass.

The second hit is: https://github.com/svagner/vixie-cron/blob/master/cron.c - that's vixie-cron, not vixie/cron.  Oops.

I'm going to let this drop - we've both put more time into it than it's worth.

Sorry about that.
